### PR TITLE
#10036 Replace codename Merge with Bellatrix (2nd step)

### PR DESCRIPTION
**Pull request by leolara**
**What type of PR is this?**

Refactoring

**What does this PR do? Why is it needed?**

Replace codename Merge with Bellatrix, focusing on proto package

**Which issues(s) does this PR fix?**

Partially #10036

**Other notes for review**

I had to manually fix three files generated from protobuf, for some reason ./hack/update-go-pbs.sh adds a silly syntax error to these files

### CODE REVIEW

**Comment by leolara** [comment]
@terencechain after rebase there is a protobuf compilation error, but bazel hides the underlaying error

**Comment by leolara** [comment]
@terencechain please check

**Comment by leolara** [comment]
After this have been approved I think I can fix the rest in another PR

File: proto/prysm/v1alpha1/wrapper/beacon_block.go
Comment (potuz) [inline proto/prysm/v1alpha1/wrapper/beacon_block.go:114]:
```suggestion
	return nil, errors.New("unsupported Bellatrix block")
```
This is particularly optional cause altair just above should be Altair to me being a proper name, but that's not part of this PR and other members may disagree on form here.

File: proto/prysm/v1alpha1/wrapper/beacon_block.go
Comment (potuz) [inline proto/prysm/v1alpha1/wrapper/beacon_block.go:577]:
```suggestion
// bellatrixSignedBeaconBlock is a convenience wrapper around a Bellatrix beacon block
```

File: proto/prysm/v1alpha1/wrapper/beacon_block.go
Comment (potuz) [inline proto/prysm/v1alpha1/wrapper/beacon_block.go:584]:
```suggestion
// WrappedBellatrixSignedBeaconBlock is constructor which wraps a protobuf Bellatrix block with the block wrapper.
```

File: proto/prysm/v1alpha1/wrapper/beacon_block.go
Comment (potuz) [inline proto/prysm/v1alpha1/wrapper/beacon_block.go:684]:
```suggestion
// WrappedBellatrixBeaconBlock is constructor which wraps a protobuf Bellatrix object
```

File: validator/client/propose.go
Comment (potuz) [inline validator/client/propose.go:585]:
```suggestion
		log.Error("Not a Bellatrix block")
```

File: validator/keymanager/remote-web3signer/keymanager.go
Comment (potuz) [inline validator/keymanager/remote-web3signer/keymanager.go:134]:
```suggestion
	// TODO(#10053): Need to add support for Bellatrix blocks.
```

**Review by potuz — APPROVED** [review]
For other reviewers (if others are deemed necessary) here are some sanity checks that I have passed through this PR:
1- `git diff develop --name-only | xargs cat | grep Merge`  to catch for missed merge names in the changed files. Only few instances are found, only verbs. 
2. Same with  "merge". 
3. Checked all instances of changes in the diff to see if they made sense. (without actually checking for code logic)

I left comments throughout the PR requesting to change "bellatrix" for "Bellatrix". These are optionals. I know there's a rule about error messages being lower case, but to my own taste proper names should be capitalized. The same applies to Altair which is not the scope of this PR, and others may disagree, so I'll approve right now and leave those comments there in case someone wants to change them.

**Comment by leolara** [comment]
This seems ready to merge!! LGTM

File: beacon-chain/core/execution/upgrade.go
Comment (potuz) [inline beacon-chain/core/execution/upgrade.go:15]: Are we sure about this change? It hides an error that is unrelated to this PR and may make it harder to debug later if the function is not properly implemented. The parameter should go, or just keep the error. Incidentally. Each time you push you make stale my previous approval. The PR does look good to me

Comment (leolara) [reply beacon-chain/core/execution/upgrade.go:15]: @potuz I don't have an opinion about this, I just did it so "Deepsource" wouldn't protest. If we can merge it with the "Deepsource" error, I can revert it, no problem.

Are you suggesting to remove the parameter all the way? Or merging ignoring "Deepsource"?

Comment (potuz) [reply beacon-chain/core/execution/upgrade.go:15]: I would suggest to leave the error as is since this is not the scope of the PR, we can wait until Monday and let @terencechain chime in, since he'll write the PR implementing the upgrade, but either way, hiding the error i think shouldn't be the best option

Comment (terencechain) [reply beacon-chain/core/execution/upgrade.go:15]: +1

Let's leave the error as it is

Comment (leolara) [reply beacon-chain/core/execution/upgrade.go:15]: ok

**Comment by leolara** [comment]
@terencechain removed the error correction. Let me know if we need anything else to merge this? :-)

**Review by rkapka — DISMISSED** [review]
Approving changes to `state` package.

**Comment by leolara** [comment]
I just rebased to make in sync with develop.

Are we merging this? @rkapka @terencechain @potuz 

I think it does not let me even if approved because it does not pass the "deepsource: go" step

**Comment by potuz** [comment]
> I just rebased to make in sync with develop.
> 
> Are we merging this? @rkapka @terencechain @potuz
> 
> I think it does not let me even if approved because it does not pass the "deepsource: go" step

It's fine the Deepsource error, we can merge with that. The problem is that each time you push changes, you make the previous approval stale. There's been like 10 pushes since I last approved this, will need to review them. I suppose the same applies to @rkapka's review

**Comment by leolara** [comment]
Sorry, I will not rebase anymore. I didn't know it would cause trouble.

I don't like to have many "merge develop" into the history, so I usually do that. But I will not do this here anymore

**Comment by terencechain** [comment]
> I don't like to have many "merge develop" into the history, so I usually do that. But I will not do this here anymore

Everything gets squashed into 1 commit when we merge into `develop` so don't worry about the # of commits that you do

**Review by rkapka — APPROVED** [review]

**Comment by rkapka** [comment]
Let's wait for approval from @terencechain or @potuz before merging.

**Review by potuz — APPROVED** [review]
As far as I can tell these changes are fine

**Comment by leolara** [comment]
@potuz it does not automerge because of the deepsource error. What should we do?

**Comment by nisdas** [comment]
@leolara I just started the CI build for your PR, it should merge soon if there are any issues. Deepsource isn't a requirement for a successful merge of the PR.

**Comment by leolara** [comment]
Thanks @nisdas I am just starting to learn your process :-)

**Pull request merged**
